internal/datanode/compactor.go

```diff
@@ -41,11 +43,14 @@ var (
 	errIllegalCompactionPlan   = errors.New("compaction plan illegal")
 	errTransferType            = errors.New("transfer intferface to type wrong")
 	errUnknownDataType         = errors.New("unknown shema DataType")
+	errContext                 = errors.New("context done or timeout")
 )
 
 type iterator = storage.Iterator
 
 type compactor interface {
+	add()
+	done()
```

Comment:
logic ok, but `add` and `done` is not meaningful name of function. Maybe `start(runningState)`/ `complete(completedState)` is better?